### PR TITLE
add admin mode

### LINKED ISSUE #40: Support enable & disable advance/admin mode
There should be a check button to enable/disable advance/admin mode in which it is possible to see `x_editor_hidden` and edit `x_editor_disabled` fields.

### PR DESCRIPTION
**Pull request by harunurhan**
* Fixes #40 

UI to be discussed.

admin-mode

### CODE REVIEW

**Comment by jmartinm** [comment]
LGTM.

For the UI, I would for the moment leave it as you have it with a label next to the checkbox. `Unlock protected fields`

**Comment by harunurhan** [comment]
Updated:

screen shot 2017-01-23 at 17 41 22

**Comment by jmartinm** [comment]
The issue I have with `Enable Admin Mode` is that it is not self explanatory, you will have to click to see what happens.

We can leave it like that but add a tooltip on hover that says: "Allows editing all fields (use with care)".

**Comment by zzacharo** [comment]
This LGTM too. Also maybe the ui of this admin could be part of the record-editor on the toolbar. So, we could have a global exported admin input on our json-editor app.

**Comment by jmartinm** [comment]
> This LGTM too. Also maybe the ui of this admin could be part of the record-editor on the toolbar. So, we could have a global exported admin input on our json-editor app. And have some flag like cancel_hidden_disable_filters here instead of adminMode.

Yes, that's a good idea since this is something quite INSPIRE-specific and probably another installation using the editor would not want to allow people to edit everything.

Can we have a config variable that enables/disables this feature and we set it to `true` on the record-editor?

**Comment by jmartinm** [comment]
Also for it to be ready note [the change](https://github.com/inveniosoftware-contrib/ng2-json-editor/pull/151#issuecomment-274545785) suggested above.

**Comment by harunurhan** [comment]
@zzacharo I think it should be here at least for now, but we can have a config to disable admin mode which will never display the checkbox.

@jmartinm addressed your change

**Comment by harunurhan** [comment]
Now your comment, so will add the config.

**Comment by harunurhan** [comment]
@jmartinm added the config, and rebased it.

Also realized that we should really stop using typings because for config type, it needs be repeated for example.

File: README.md [hunk 1]
```diff
@@ -43,6 +43,7 @@ Please have a look at [example app](./example/app)
 | *schemaOptions*      | Object | Enriches JsonSchema with custom properties that `ng2-json-editor` understands. See [possible values](#schemaOptions).|
 | *previews*      | Array | Configures previews for the document. See [configuration syntax](#previews).|
 | *shortcuts*     | Object | Configures shortcuts for the document. See [shortcuts syntax](#shortcuts).|
+| *enableAdminModeSwitch*      | boolean=false | If set, displays the switch that can enable admin mode where anything field can be edited.|
```
Comment (jmartinm) [inline README.md]: `anything field` -> `any field`

File: src/json-editor.component.html [hunk 2]
```diff
@@ -2,6 +2,10 @@
   <div class="col-md-2 left">
     <tree-menu [record]="_record" [schema]="schema"></tree-menu>
     <add-field-dropdown [fields]="keys" [pathString]="''" (onFieldAdd)="onFieldAdd($event)" [schema]="schema.properties">Add field</add-field-dropdown>
+    <div *ngIf="config.enableAdminModeSwitch"class="admin-mode" tooltip="Allows editing all fields (use with care)">
```
Comment (jmartinm) [inline src/json-editor.component.html]: space missing after `*ngIf="config.enableAdminModeSwitch"`

**Comment by harunurhan** [comment]
@jmartinm addressed the comments

**Pull request merged**